client/my-sites/sidebar-unified/item.jsx

```diff
@@ -8,21 +8,32 @@
 /**
  * External dependencies
  */
-import React from 'react';
+import React, { memo } from 'react';
 import { useSelector } from 'react-redux';
 import PropTypes from 'prop-types';
 
 /**
  * Internal dependencies
  */
 
-import { getSelectedSiteId } from 'state/ui/selectors';
-import SidebarItem from 'layout/sidebar/item';
-import SidebarCustomIcon from 'layout/sidebar/custom-icon';
-import StatsSparkline from 'blocks/stats-sparkline';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+import SidebarItem from 'calypso/layout/sidebar/item';
+import SidebarCustomIcon from 'calypso/layout/sidebar/custom-icon';
+import StatsSparkline from 'calypso/blocks/stats-sparkline';
 
 const onNav = () => null;
 
+const MySitesSidebarUnifiedStatsSparkline = memo( ( { slug } ) => {
+	const selectedSiteId = useSelector( getSelectedSiteId );
+	const isStats = typeof slug === 'string' && slug.includes( '-comstats' );
+
+	if ( isStats && selectedSiteId ) {
+		return null;
+	}
+
+	return <StatsSparkline className="sidebar-unified__sparkline" siteId={ selectedSiteId } />;
+} );
```

Comment:
This is a reasonably costly component to render so by memorizing it we only cause it to re-render if/when the `slug` prop changes (or the `selectedSiteId` changes).